Backend/src/bdd/java/com/example/Backend/stepDefination/LoginStepDefination.java

```diff
@@ -0,0 +1,79 @@
+package com.example.Backend.stepDefination;
+
+import com.example.Backend.model.UserLogin;
+import com.example.Backend.model.UserRegister;
+import com.example.Backend.model.UserResponse;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import lombok.extern.slf4j.Slf4j;
+import net.serenitybdd.core.Serenity;
+import org.junit.jupiter.api.Assertions;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.ResponseEntity;
+
+@Slf4j
+public class LoginStepDefination extends StartupHook{
+    UserRegister tester = new UserRegister();
+    UserLogin testLogin = new UserLogin();
+    TestRestTemplate restTemplate = new TestRestTemplate();
+    ResponseEntity<UserResponse> responseRegister, responseLogin;
+    ResponseEntity<UserLogin> result;
+
+    @Given("User calls login endpoint for valid user")
+    public void userCallsLoginEndpointForValidUser() {
+        tester.setUsername("user2");
+        tester.setEmail("tester2@gmail.com");
+        tester.setPassword("Pass@123");
+        responseRegister = restTemplate.postForEntity("http://localhost:8083/user/register", tester, UserResponse.class);
+        testLogin.setUsername(tester.getUsername());
+        testLogin.setPassword(tester.getPassword());
+    }
+
+    @When("System processes login request for valid user")
+    public void systemProcessesLoginRequestForValidUser() {
+        log.info("Processing login request");
+        responseLogin = restTemplate.postForEntity("http://localhost:8083/user/login", testLogin, UserResponse.class);
+        Serenity.setSessionVariable("response").to(responseLogin.getBody());
+    }
+
+    @Then("Validate response from the login endpoint for valid user")
+    public void validateResponseFromTheLoginEndpointForValidUser() {
+        Assertions.assertEquals(UserResponse.builder().Message("User Logged in: " + tester.getUsername()).build(), Serenity.getCurrentSession().get("response"));
+    }
+
+    @Given("User calls login endpoint for invalid user")
+    public void userCallsLoginEndpointForInvalidUser() {
+        testLogin.setUsername("nouser");
+        testLogin.setPassword("Password");
+    }
+
+    @When("System processes login request for invalid user")
```

Comment:
There is a way to extract keyword from sentences in feature file and use it in variable in stepDefinattions. 
Same sentence can be used for valid and invalid username

Read - https://stackoverflow.com/questions/52421634/how-to-pass-data-from-the-feature-file-to-step-definition-file-using-regular-exp